homeassistant/components/dremel_3d_printer/__init__.py

```diff
@@ -0,0 +1,156 @@
+    domain_config = config[DOMAIN]
+
+    for conf in domain_config:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_USER},
+                data={
+                    CONF_HOST: conf[CONF_HOST],
+                },
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up Dremel 3D Printer from a config entry."""
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+
+    try:
+        api = await hass.async_add_executor_job(
+            Dremel3DPrinter, config_entry.data[CONF_HOST]
+        )
+
+    except (ConnectTimeout, HTTPError) as exc:
+        raise ConfigEntryNotReady(
+            f"Unable to connect to Dremel 3D Printer: {exc}"
+        ) from exc
+    except Exception as exc:
+        raise ConfigEntryNotReady(
+            f"Unknown error connecting to Dremel 3D Printer: {exc}"
+        ) from exc
+
+    coordinator = Dremel3DPrinterDataUpdateCoordinator(
+        hass, f"{DOMAIN}-{config_entry.entry_id}", api, SCAN_INTERVAL.seconds
+    )
+    await coordinator.async_config_entry_first_refresh()
+    hass.data[DOMAIN][config_entry.entry_id] = coordinator
+    hass.config_entries.async_setup_platforms(config_entry, PLATFORMS)
+    await async_setup_services(hass)
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload Dremel config entry."""
+
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        del hass.data[DOMAIN]
+
+    return unload_ok
+
+
+class Dremel3DPrinterDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching Dremel 3D Printer data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        name: str,
+        api: Dremel3DPrinter,
+        interval: int,
+    ) -> None:
+        """Initialize Dremel 3D Printer data update coordinator."""
+        DataUpdateCoordinator.__init__(
+            self,
+            hass,
+            _LOGGER,
+            name=name,
+            update_interval=timedelta(seconds=interval),
+        )
+        self.printer_offline = False
+        self.api = api
+
+    async def _async_update_data(self) -> dict[str, Any]:
+        """Update data via APIs."""
+        try:
+            await self.hass.async_add_executor_job(self.api.refresh)
+        except RuntimeError:
+            if not self.printer_offline:
+                _LOGGER.debug("Unable to refresh printer information: Printer offline")
+                self.printer_offline = True
+        else:
+            self.printer_offline = False
+
+        return {
+            "last_read_time": dt_util.utcnow(),
+        }
+
+
+class Dremel3DPrinterDeviceEntity(
+    CoordinatorEntity[Dremel3DPrinterDataUpdateCoordinator], Entity
+):
+    """Defines a Dremel 3D Printer device entity."""
+
```

Comment:
New integrations are now required to use the new entity naming style. Please see: https://developers.home-assistant.io/docs/core/entity/#entity-naming